### PR TITLE
Hook up the lisp extension automation test on Github CI

### PR DESCRIPTION
**Pull request by nigma456**
With this PR I have added the test to Github CI and change the OS to Windows

##### Abstractions

With more and more coming submissions we should make sure there are no regressions caused so we hooked the automation test to Github CI which will run with every submission.

At the same time due to the limitaion of vscode on Linux we decide to run the test on Windows on Github CI.

Add codeql-config.yml to ignore some of the path to avoid unnecessary scanning for code quality check.

I have also modified some of the test cases.

##### Tests performed
Run passed on local test and Github CI.

### CODE REVIEW

File: extension/src/test/suite/DocumentContainer.test.ts [hunk 1]
```diff
@@ -35,7 +37,7 @@ suite("LispParser.DocumentContainer Tests", function () {
 	test("DocumentExpression using index", function () {		
 		try {
 			const expectation = '(= (length retList) 1)';
-			const doc = ReadonlyDocument.open(project_path); 						
+			const doc = ReadonlyDocument.open(lispFileTest);
```
Comment (LanDong-adsk) [inline extension/src/test/suite/DocumentContainer.test.ts:40]: code scanning said the diff is unused. please remove it. same at below places.

File: extension/src/test/suite/index.ts [hunk 2]
```diff
@@ -2,15 +2,7 @@ import * as path from 'path';
...
-import * as fs from 'fs-extra';
-import * as baseConfig from "@istanbuljs/nyc-config-typescript";
-
-import 'ts-node/register';
-import 'source-map-support/register';
-
-const NYC = require('nyc');
-
-export async function run(): Promise<void> {
+export function run(): Promise<void> {
```
Comment (LanDong-adsk) [inline extension/src/test/suite/index.ts:5]: this is overwritting Sen's previous change. Please merge first.

**Review by LanDong-adsk — CHANGES_REQUESTED** [review]

File: extension/src/test/suite/index.ts (hunk 2 above)
Comment (nigma456) [reply extension/src/test/suite/index.ts:5]: OK, I will take a look

File: extension/src/test/suite/DocumentContainer.test.ts (hunk 1 above)
Comment (nigma456) [reply extension/src/test/suite/DocumentContainer.test.ts:40]: Ok, I will remove the diff variable.

File: package.json [hunk 3]
```diff
@@ -627,6 +627,7 @@
 		"chai": "^4.2.0",
 		"del": "^4.1.1",
 		"event-stream": "^4.0.1",
+		"fs": "0.0.1-security",
```
Comment (Sen-real) [inline package.json:630]: what is this for?

Comment (nigma456) [reply package.json]: Remove from package.json.

File: package.json [hunk 4]
```diff
@@ -615,7 +615,8 @@
 	"scripts": {
 		"compile": "tsc -p ./",
 		"watch": "tsc -p -watch ./",
-		"test": "npm run compile && node ./out/test/runTest.js"
+		"test": "npm run compile && node ./out/test/runTest.js",
+		"codecoverage": "npm run compile && node ./out/test/runCodeCoverage.js"
```
Comment (LanDong-adsk) [inline package.json:619]: what is this file? if it is not ready, we can do it later in another PR.

Comment (Sen-real) [inline package.json:619]: thanks a lot! is this file uploaded?

Comment (LanDong-adsk) [reply package.json:619]: btw, I see even with code coverage, the test still runs very quickly. So this is not high priority as of now. We can just run the tests with coverage support.

Comment (nigma456) [reply package.json:619]: Yes, we can run the tests with coverage with this PR, and I will submit a new PR to decouple the code coverage to a separate command.

Comment (nigma456) [reply package.json]: I will remove this line from this PR

**Review by LanDong-adsk — APPROVED** [review]

**Pull request merged**